quarkus.jdt.ext/com.redhat.microprofile.jdt.quarkus/src/main/java/com/redhat/microprofile/jdt/internal/quarkus/scheduler/java/QuarkusSchedulerASTVisitor.java

```diff
@@ -0,0 +1,142 @@
+package com.redhat.microprofile.jdt.internal.quarkus.scheduler.java;
+
+import static org.eclipse.lsp4mp.jdt.core.utils.AnnotationUtils.getAnnotationMemberValueExpression;
+import static org.eclipse.lsp4mp.jdt.core.utils.AnnotationUtils.isMatchAnnotation;
+
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.core.dom.Annotation;
+import org.eclipse.jdt.core.dom.Expression;
+import org.eclipse.jdt.core.dom.MethodDeclaration;
+import org.eclipse.jdt.core.dom.NormalAnnotation;
+import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.eclipse.lsp4mp.jdt.core.java.diagnostics.JavaDiagnosticsContext;
+import org.eclipse.lsp4mp.jdt.core.java.validators.JavaASTValidator;
+import org.eclipse.lsp4mp.jdt.core.utils.JDTTypeUtils;
+
+import com.redhat.microprofile.jdt.internal.quarkus.QuarkusConstants;
+import com.redhat.microprofile.jdt.internal.quarkus.scheduler.SchedulerErrorCodes;
+import com.redhat.microprofile.jdt.internal.quarkus.scheduler.SchedulerUtils;
+
+public class QuarkusSchedulerASTVisitor extends JavaASTValidator {
+
+	private static Logger LOGGER = Logger.getLogger(QuarkusSchedulerASTVisitor.class.getName());
+
+	public QuarkusSchedulerASTVisitor() {
+		super();
+	}
+
+	@Override
+	public boolean isAdaptedForDiagnostics(JavaDiagnosticsContext context, IProgressMonitor monitor)
+			throws CoreException {
+		IJavaProject javaProject = context.getJavaProject();
+		return JDTTypeUtils.findType(javaProject, QuarkusConstants.SCHEDULED_ANNOTATION) != null;
+	}
+
+	@Override
+	public boolean visit(MethodDeclaration node) {
+		try {
+			List modifiers = node.modifiers();
+			for (Object modifier : modifiers) {
+				if (modifier instanceof Annotation) {
+					Annotation annotation = (Annotation) modifier;
+					if (isMatchAnnotation(annotation, QuarkusConstants.SCHEDULED_ANNOTATION)) {
+						validateScheduledAnnotation(node, (NormalAnnotation) annotation);
+					}
+				}
+			}
+		} catch (JavaModelException e) {
+			LOGGER.log(Level.WARNING, "An exception occurred when attempting to validate the annotation marked method");
+		}
+		super.visit(node);
+		return true;
+	}
+
+	/**
+	 * Checks if the given method declaration has a @Scheduled annotation, and if
+	 * so, provides diagnostics it's member(s)
+	 *
+	 * @param node       The method declaration to validate
+	 * @param annotation The @Scheduled annotation
+	 * @throws JavaModelException
+	 */
+	private void validateScheduledAnnotation(MethodDeclaration node, NormalAnnotation annotation)
+			throws JavaModelException {
+		Expression cronExpr = getAnnotationMemberValueExpression(annotation,
+				QuarkusConstants.SCHEDULED_ANNOTATION_CRON);
+		if (cronExpr != null) {
+			String cronValue = cronExpr.resolveConstantExpressionValue().toString();
```

Comment:
Need to watch out for when `resolveConstantExpressionValue()` returns null. This can happen the instant a user creates the member in an annotation and it has no expression. Eg. `@Scheduled(cron = )`

```
Returns:
the constant expression value, or null if this expression has no constant expression value or if bindings were not requested when the AST was created
```